pages/check_availability.tsx

```diff
@@ -40,6 +40,38 @@ export default function CheckAvailability() {
       });
     }
   };
+//==============================================================================
+// function for getting user district and displaying CVCs
+function getUserLocation() {
+  const success= async(pos)=>{
+    const latitude = pos.coords.latitude;
+    const longitude = pos.coords.longitude;
+    try {
+      const result = await fetch(`https://api.opencagedata.com/geocode/v1/json?q=${latitude}+${longitude}&key=${process.env.NEXT_PUBLIC_GEOCODING_API_KEY}`);
+      const resultJSON = await result.json();
+      const district= resultJSON.results[0].components.county;
+      setSearch(district);  
+      push({
+        pathname: "/availability_results",
+        query: {
+          district: district,
+        },
+      });
```

Comment:
IMO we should make the query using the Pincode/postal code

The API currently query for the districts is case sensitive, it only supports the following Districts, and they have to be word to word the same (This might change if we bring something more advanced in the future)

https://github.com/IndiaVaccineIN/india-vaccine-frontend/blob/master/api/district.ts

```suggestion
      push({
        pathname: "/availability_results",
        query: {
          pincode: search,
        },
      });
```